src/main/java/com/redhat/developers/msa/ola/tracing/TracerResolver.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2016 Juraci Paixão Kröhling
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.developers.msa.ola.tracing;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import java.util.logging.Logger;
+
+import io.opentracing.Tracer;
+
+/**
+ * This is a helper class, that serves as an example of what could be done as a microservices library
+ * for detecting the appropriate Tracer based on the environment. On this case, we explicitly know about
+ * two tracers, BraveTracer and APMTracer.
+ *
+ * For the Hello World MSA, we might move this to a common module in the future.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@SuppressWarnings("Duplicates")
```

Comment:
This `Duplicates` is here because we have very similar codes in other modules. There's no "commons" on MSA yet (at least, not that I know of). Would it be worth creating one, just for this tracing part? Or is it OK to have duplicate code?